examples/navigation/src/main.rs

```diff
@@ -0,0 +1,80 @@
+#![allow(rustdoc::missing_crate_level_docs)]
+
+use eframe::{App, Frame, egui};
+use egui::{Color32, Context, Key, Modifiers, PointerButton, Vec2b};
+use egui_plot::{
+    Line, Plot, TooltipOptions, {BoxZoomConfig, NavigationConfig, ResetBehavior, ZoomConfig},
+};
+
+fn main() -> eframe::Result<()> {
+    eframe::run_native(
+        "Line::new_xy + custom navigation",
+        eframe::NativeOptions::default(),
+        Box::new(|_| Ok(Box::new(Demo::new()))),
+    )
+}
+
+struct Demo {
+    xs: Vec<f64>,
+    f1: Vec<f64>,
+    f2: Vec<f64>,
+}
+
+impl Demo {
+    fn new() -> Self {
+        let n = 500;
+        let xs: Vec<f64> = (0..n).map(|i| i as f64 * 0.02).collect();
+        let f1: Vec<f64> = xs.iter().map(|&t| t.sin()).collect();
+        let f2: Vec<f64> = xs
+            .iter()
+            .map(|&t| (t * 0.6 + 0.8).sin() * 0.8 + 0.2)
+            .collect();
+
+        Self { xs, f1, f2 }
+    }
+}
+
+impl App for Demo {
+    fn update(&mut self, ctx: &Context, _frame: &mut Frame) {
+        egui::CentralPanel::default().show(ctx, |ui| {
+            ui.heading("Line::new_xy + completely custom navigation");
+
+            let nav = NavigationConfig::default()
```

Comment:
High-level comment: with config it should be clear what is being set without the need to open source code.
When reading it on github that is not the case